test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,952 @@
+    constructor() {
+        this.class = Car;
+        this.typeName = 'Car';
+    }
+
+    read(reader) {
+        compactSerializerUsed = true;
+        const name = reader.readString('name');
+        const price = reader.readInt32('price');
+
+        return new Car(name, price);
+    }
+
+    write(writer, instance) {
+        compactSerializerUsed = true;
+        writer.writeString('name', instance.name);
+        writer.writeInt32('price', instance.price);
+    }
+}
+
+class DummyEntryProcessor { }
+
+class DummyEntryProcessorSerializer {
+    constructor() {
+        this.class = DummyEntryProcessor;
+        this.typeName = 'DummyEntryProcessor';
+    }
+
+    read() {
+        compactSerializerUsed = true;
+        return new DummyEntryProcessor();
+    }
+
+    write() {
+        compactSerializerUsed = true;
+    }
+}
+
+const portableFactory = classId => {
+    if (classId === 2) {
+        return new DummyEntryProcessor();
+    }
+};
+
+class CompactReturningAggregator {
+}
+
+class CompactReturningAggregatorSerializer {
+    constructor() {
+        this.class = CompactReturningAggregator;
+        this.typeName = 'CompactReturningAggregator';
+    }
+
+    read() {
+        compactSerializerUsed = true;
+        return new CompactReturningAggregator();
+    }
+
+    write() {
+        compactSerializerUsed = true;
+    }
+}
+
+class CompactPredicate {
+}
+
+class CompactPredicateSerializer {
+    constructor() {
+        this.class = CompactPredicate;
+        this.typeName = 'CompactPredicate';
+    }
+
+    read() {
+        compactSerializerUsed = true;
+        return new CompactPredicate();
+    }
+
+    write() {
+        compactSerializerUsed = true;
+    }
+}
+
+class Inner {
+    constructor(stringField) {
+        this.stringField = stringField;
+    }
+}
+
+class InnerSerializer {
+    constructor() {
+        this.class = Inner;
+        this.typeName = 'Inner';
+    }
+
+    read(reader) {
+        compactSerializerUsed = true;
+        const stringField = reader.readString('string_field');
+        return new Inner(stringField);
+    }
+
+    write(writer, instance) {
+        compactSerializerUsed = true;
+        writer.writeString('string_field', instance.stringField);
+    }
+}
+
+class Outer {
+    constructor(intField, innerField) {
+        this.intField = intField;
+        this.innerField = innerField;
+    }
+}
+
+class OuterSerializer {
+    constructor() {
+        this.class = Outer;
+        this.typeName = 'Outer';
+    }
+
+    read(reader) {
+        compactSerializerUsed = true;
+        const intField = reader.readInt32('int_field');
+        const innerField = reader.readCompact('inner_field');
+        return new Outer(intField, innerField);
+    }
+
+    write(writer, instance) {
+        compactSerializerUsed = true;
+        writer.writeInt32('int_field', instance.intField);
+        writer.writeCompact('inner_field', instance.innerField);
+    }
+}
+
+class CompactPagingPredicate {
```

Comment:
after looking at the implementations where we use paging predicate, it seems that we don't support custom paging predicate implementations. the users has to use our own implementation. so, we should get rid of this and simply use Predicates.paging(new CompactPredicate()). I will do the same for Python